vehicle/src/Vehicle/Backend/LossFunction/Compile.hs

```diff
@@ -137,78 +73,84 @@ compileDecl d =
     normalisationError currentPass "postulates"
 
   V.DefFunction _ ident _ expr -> do
-    (propertyCtx, _) <- ask
-    let isProperty = Map.member ident propertyCtx
-    if not isProperty
-      -- If it's not a property then we can discard it as all applications
-      -- of it should have been normalised out by now.
-      then return Nothing
-      else do
-        expr' <- compileExpr expr
-        logDebug MaxDetail ("loss-declaration " <> prettySimple expr)
-        return (Just (DefFunction (nameOf ident) expr'))
+    expr' <- compileExpr t expr
+    logDebug MaxDetail ("loss-declaration " <> prettySimple expr)
+    return (DefFunction (nameOf ident) expr')
 
 currentPass :: Doc a
 currentPass = "compilation to loss functions"
 
-compileArg :: MonadCompile m => V.CheckedArg -> m LExpr
-compileArg arg = compileExpr (V.argExpr arg)
-
-compileLiteral :: V.Literal -> Double
-compileLiteral = \case
-  V.LUnit{}    -> developerError "LUnit"
-  V.LBool{}    -> developerError "LBool"
-  V.LIndex _ e -> fromIntegral e
-  V.LNat     e -> fromIntegral e
-  V.LInt     e -> fromIntegral e
-  V.LRat     e -> fromRational e
-
-compileExpr :: MonadCompile m => V.CheckedExpr -> m LExpr
-compileExpr e = showExit $ do
+compileArg :: MonadCompile m => DifferentialLogicImplementation -> V.CheckedArg -> m LExpr
+compileArg t arg = compileExpr t (V.argExpr arg)
+
+-- |helper function for compiling Literals
+compileLiteral :: DifferentialLogicImplementation -> V.Literal -> Double
+compileLiteral t l = case l of
+  V.LUnit{}        -> developerError "Loss Function should not encounter LUnit"
+  V.LBool    True  -> compileTrue t
+  V.LBool    False -> compileFalse t
+  V.LIndex _ e     -> fromIntegral e
+  V.LNat     e     -> fromIntegral e
+  V.LInt     e     -> fromIntegral e
+  V.LRat     e     -> fromRational e
+
+-- |helps compile a name from DBBinding, even if there is no name given
+compileDBBinding :: Maybe Name -> Name
+compileDBBinding = fromMaybe "No_name"
+
+--compileMaybeNot :: m LExpr -> m LExpr
+--compileMaybeNot not = case not of
+--  Nothing -> (lowerNot (argExpr e1))
+
+-- |compile a property or single expression
+compileExpr :: MonadCompile m => DifferentialLogicImplementation -> V.CheckedExpr -> m LExpr
+compileExpr t e = showExit $ do
   e' <- showEntry e
   case e' of
     --logical operatives
-    V.NotExpr     _ [e1]     -> Negation <$> compileArg e1
-    V.AndExpr     _ [e1, e2] -> Min <$> compileArg e1 <*> compileArg e2
-    V.OrExpr      _ [e1, e2] -> Max <$> compileArg e1 <*> compileArg e2
-    V.ImpliesExpr _ [e1, e2] -> Max <$> (Negation <$> compileArg e1) <*> compileArg e2
+    V.NotExpr     _ [e1]     -> case compileNot t of
+                                  Nothing -> compileExpr t (lowerNot (argExpr e1))--compileNot t <$> compileExpr t (lowerNot (argExpr e1))
+                                  Just f -> f <$> compileArg t e1
```

Comment:
Delete remaining comment, and we should probably decrease the indentation of the cases to only 2 spaces beyond the `V.NotExpr`